From 7a00cfe48241f5ca2b8fd4d54c93cb88f1c1b99d Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Tue, 20 Feb 2024 18:07:29 +0100
Subject: [PATCH 576/954] feat(plat-stm32mp2): remove stm32_iwdg_shadow_update

Remove the function stm32_iwdg_shadow_update(), only called by
stm32_iwdg driver() to force the shadow value of OTP HCONF1.
It is not required for STM32MP25 BL2, the shadow value for this
OTP is only managed in OP-TEE.

This function will be also removed for STM32MP1 and the IWDG
driver will be cleanup when the same behavior will be coded
in OP-TEE for this platform.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git beb806fae0e3f63c4425c8335d7498dfe787be91 feat(plat-stm32mp2): remove stm32_iwdg_shadow_update]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Id985df96b3454830ac508f37923e79ad613c66fb
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/361447
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/368505
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
Tested-by: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 plat/st/stm32mp2/stm32mp2_private.c | 27 ---------------------------
 1 file changed, 27 deletions(-)

diff --git a/plat/st/stm32mp2/stm32mp2_private.c b/plat/st/stm32mp2/stm32mp2_private.c
index c1a999e3b7..b07838455a 100644
--- a/plat/st/stm32mp2/stm32mp2_private.c
+++ b/plat/st/stm32mp2/stm32mp2_private.c
@@ -579,33 +579,6 @@ uint32_t stm32_iwdg_get_otp_config(uint32_t iwdg_inst)
 #if defined(IMAGE_BL2)
 uint32_t stm32_iwdg_shadow_update(uint32_t iwdg_inst, uint32_t flags)
 {
-	uint32_t otp_value;
-	uint32_t otp_idx;
-	uint32_t result;
-
-	if (stm32_get_otp_index(HCONF1_OTP, &otp_idx, NULL) != 0U) {
-		panic();
-	}
-
-	if (stm32_get_otp_value_from_idx(otp_idx, &otp_value) != 0U) {
-		panic();
-	}
-
-	if ((flags & IWDG_DISABLE_ON_STOP) != 0U) {
-		otp_value |= BIT(iwdg_inst + HCONF1_OTP_IWDG_FZ_STOP_POS);
-	}
-
-	if ((flags & IWDG_DISABLE_ON_STANDBY) != 0U) {
-		otp_value |= BIT(iwdg_inst + HCONF1_OTP_IWDG_FZ_STANDBY_POS);
-	}
-
-	result = stm32_otp_write(otp_value, otp_idx);
-	if (result != BSEC_OK) {
-		return result;
-	}
-
-	/* Do not sticky lock OTP_IWDG (read and write) */
-
 	return BSEC_OK;
 }
 #endif
-- 
2.53.0

